libeos-updater-util/avahi-service-file.c

```diff
@@ -38,65 +41,131 @@ const gchar * const eos_avahi_v1_ostree_path = "eos_ostree_path";
 const gchar * const eos_avahi_v1_head_commit_timestamp = "eos_head_commit_timestamp";
 
 static gchar *
-txt_records_to_string (const gchar **txt_records)
+txt_records_to_string (GVariant *txt_records)
 {
+  GVariantIter iter;
   g_autoptr(GString) str = NULL;
-  gsize idx;
+  GVariant *txt_value;
+  const gchar *txt_key;
+
+  g_assert (g_variant_type_equal (g_variant_get_type (txt_records),
```

Comment:
This is a static function, so I think that g_assert is fine. If it fires, then it means I have made a mistake somewhere, as txt records are constructed entirely in this file.

Thanks for the hint about `g_variant_is_of_type` - I can see it is not exactly the same as using `g_variant_type_equal`, but it shouldn't matter in our case.